Apps/WebClient/src/ClientApp/src/components/timeline/linearTimeline.vue

```diff
@@ -120,10 +121,54 @@ export default class LinearTimelineComponent extends Vue {
             self.setPageFromDate(eventDate);
         });
 
+        this.eventBus.$on(
+            EventMessageName.TimelineCurrentDateUpdated,
+            function (currentDate: Date) {
+                self.goToPageByCurrentDate(currentDate);
+            }
+        );
+
         window.addEventListener("resize", this.handleResize);
         this.handleResize();
     }
 
+    private goToPageByCurrentDate(currentDate: Date) {
+        const selectedYearMonth = moment(currentDate).format("YYYY-MM");
+        let currentPage = 1;
+        let currentEntryIndex = 1;
+        let i = 0;
+        let foundLastEntryOfSelectedMonth = false;
+        const timelineEntriesLenght = this.timelineEntries.length;
+        // scan from the most recent entries to the older ones
+        while (i < timelineEntriesLenght && !foundLastEntryOfSelectedMonth) {
+            let entry = this.timelineEntries[i];
+            if (
+                entry.date !== undefined &&
+                entry.date.toString().indexOf(selectedYearMonth) == 0
+            ) {
+                console.log("found the last entry of the selected month");
+                foundLastEntryOfSelectedMonth = true;
```

Comment:
instead of iterating over the entries here, we can select the first entry in the month calendar view. After that, we can use the same logic to find the date on the list and move the page there. The change would be to split that function between the focus and just the page change

Reply:
I see what you mean, only thing that I can see is the order of entries i.e. calendar shows most recent entry last, but timeline shows them first. We can discuss this and other comments above on a call tomorrow to make sure.